mmv1/third_party/terraform/services/netapp/resource_netapp_storage_pool_test.go.erb

```diff
@@ -0,0 +1,254 @@
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderBetaFactories(t),
+		CheckDestroy:             testAccCheckNetappstoragePoolDestroyProducer(t),
+		ExternalProviders: map[string]resource.ExternalProvider{
+			"time": {},
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: testAccNetappstoragePool_FlexRegionalStoragePoolCreateExample_full(context),
+			},
+			{
+				ResourceName:            "google_netapp_storage_pool.test_pool",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"location", "name", "labels", "terraform_labels"},
+			},
+			{
+				Config: testAccNetappstoragePool_FlexRegionalStoragePoolCreateExample_switchZone(context),
+			},
+			{
+				ResourceName:            "google_netapp_storage_pool.test_pool",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"location", "name", "labels", "terraform_labels"},
+			},
+			{
+				Config: testAccNetappstoragePool_FlexRegionalStoragePoolCreateExample_switchBackZone(context),
+			},
+			{
+				ResourceName:            "google_netapp_storage_pool.test_pool",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"location", "name", "labels", "terraform_labels"},
+			},
+		},
+	})
+}
+
+func testAccNetappstoragePool_FlexRegionalStoragePoolCreateExample_full(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_netapp_storage_pool" "test_pool" {
+  provider = google-beta
+  name = "tf-test-pool%{random_suffix}"
+  location = "us-east1"
+  service_level = "FLEX"
+  capacity_gib = "2048"
+  network = data.google_compute_network.default.id
+  zone = "us-east1-c"
+  replica_zone = "us-east1-b"
+}
+
+resource "time_sleep" "wait_5_minutes" {
+    depends_on = [google_netapp_storage_pool.test_pool]
+    destroy_duration = "5m"
+}
+
+data "google_compute_network" "default" {
+    provider = google-beta
+    name = "%{network_name}"
+}
+`, context)
+}
+
+func testAccNetappstoragePool_FlexRegionalStoragePoolCreateExample_switchZone(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_netapp_storage_pool" "test_pool" {
+  provider = google-beta
+  name = "tf-test-pool%{random_suffix}"
+  location = "us-east1"
+  service_level = "FLEX"
+  capacity_gib = "2048"
+  network = data.google_compute_network.default.id
+  zone = "us-east1-b"
+  replica_zone = "us-east1-c"
+}
+
+resource "time_sleep" "wait_5_minutes" {
+    depends_on = [google_netapp_storage_pool.test_pool]
+    destroy_duration = "5m"
+}
+
+data "google_compute_network" "default" {
+    provider = google-beta
+    name = "%{network_name}"
+}
+`, context)
+}
+
+func testAccNetappstoragePool_FlexRegionalStoragePoolCreateExample_switchBackZone(context map[string]interface{}) string {
+	// wait 5 minutes before executing the switchback due to api zone switch issues
+	time.Sleep(5 * time.Minute)
```

Comment:
I think this sleep is not run when we are hoping: it probably waits to start the entire test, and then still executes the step immediately. The `time_sleep` you have is more in the direction of what we will need, but I can't tell, does it do what you need? If not, we may need to tweak that part to ensure a sleep occurs before continuing with the update.

Reply:
Ok, understood. What is confusing though is that this `time.Sleep` fixed your tests. This sleep will run when the configuration is set up, but not when the tests actually run. You can see in the [debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-10946/artifacts/ec2f2cd2-3117-4cf5-aac8-19b5f773189f/recording/TestAccNetappstoragePool_FlexRegionalStoragePoolCreateExample_update.log) that there is no observable 5 min pause until 2024-06-27T11:52:01, when the deletion occurs.

Instead, I think you probably want one or two additional `"time_sleep"` resources in your config, to put artificial delays after creation (and perhaps after the first switch as well).

Reply:
Thanks @roaks3 for clarifying this, I see that the sleep at the location is not having the intended effect as you mentioned.
I have moved it to a check function which gets executed after the first switch is done and verified in the logs that the wait is happening now.
I am not sure if doing it via a sleep resource would work as intended (between the 2 switch calls) as that might depend on terraform implicit create destroy order. So I am going with adding this to the check call which gets executed post first switch.